portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -50,7 +55,11 @@ public void init() {
       PreparedQuery results = datastore.prepare(query);
 
       for (Entity entity : results.asIterable()) {
-          messages.add(entity.getProperty("comment").toString());
+          String comment = entity.getProperty("comment").toString();
```

Comment:
Important re-adjustment to have better OO going on in here:

I don't think you want to call this init method to initialize the list of messages. This is likely meant to be for initializing the actual dependencies of this class.

Here's what a better layout would be:

1. Get rid of `messages` and `count` from your class member variables.
1. Have a datastore member of the class and initialize it in place: `private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();`
1. Delete the `init` method entirely. `init` is an old Java pattern used by frameworks that don't support dependency injection through constructors. In this case, don't worry about it - it doesn't really matter here. I believe it also has access to `ServletConfig`, which is how you typically pass configuration values to the servlet. Again, not important here.
1. Everything else you had in your `init` method should go directly into your `doGet` method.

If you do this, you'll be in a much better place. In general, you want to limit your variable scope as much as possible. Notably, you never want to have a member variable that's meant to be a single-use variable for one method call.

Reply:
So my reasoning for putting it there was so that I only filled the list once and to make the code faster, so when I make the doGet call, I only have to turn the list into a JSON and send it. Does it not matter if I put the change you suggested in the doGet and it fills the list again every time?  Originally I thought the same way you did; but in order to make it faster, I thought of filling the list only the first time and then add the new items individually to the list through the doPost each time they are submitted. That way I don't have to continuously refill the entire list.